pkg/network/network.go

```diff
@@ -97,6 +97,8 @@ func inspect(addr string) (*Parameters, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed listing network interfaces: %w", err)
 	}
+
+ifLoop:
```

Comment:
thank you for this PR I think you point out correctly the logic seems wrong but I am curious, did you face any issue that made you investiage this ? if yes please link the issue for it.

Reply:
No, there is no issue here. This PR is just a small optimization